engine/version/old_records_cleaner.hpp

```diff
@@ -25,14 +25,33 @@ struct OldDataRecord {
 };
 
 struct OldDeleteRecord {
+  union RecordIndex {
+    RecordIndex(void* ptr, PointerType type)
```

Comment:
Don't union.
PointerWithTag<void, PointerType> and use access funcions such as
Type() const
HashEntryPointer() const
SkiplistNodePointer() const
RawPointer() const
to extract the pointer according to tag.

Reply:
PointerWithTag<void> is not supportted